### PR TITLE
[docs] IA3

### PR DESCRIPTION
**Pull request by stevhliu**
Adds a guide for IA3 to the docs. The `IA3Config` is pretty barebones so feel free to suggest any improvements to it!

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Thanks a lot !

**Comment by stevhliu** [comment]
Thanks! I'll wait for one more review from @BenjaminBossan when he returns from vacation before merging!

**Review by pacman100 — APPROVED** [review]
Thank you @stevhliu for adding a task guide on IA3 method! 🔥🚀✨

Left a few suggestions/comments.

File: docs/source/task_guides/ia3.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+</Tip>
+
+## Dataset
+
+You'll use the sentences_allagree subset of the [financial_phrasebank](https://huggingface.co/datasets/financial_phrasebank) dataset. This subset contains financial news with 100% annotator agreement on the sentiment label. Take a look at the [dataset viewer](https://huggingface.co/datasets/financial_phrasebank/viewer/sentences_allagree) for a better idea of the data and sentences you'll be working with.
+
+Load the dataset with the [`~datasets.load_dataset`] function. This subset of the dataset only contains a train split, so use the [`~datasets.train_test_split`] function to create a train and validation split. Create a new `text_label` column so it is easier to understand what the `label` values `0`, `1`, and `2` mean.
+
+```py
+ds = load_dataset("financial_phrasebank", "sentences_allagree")
```
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:36]: Let's add the import `from datasets import load_dataset`.

File: docs/source/task_guides/ia3.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    remove_columns=ds["train"].column_names,
+    load_from_cache_file=False,
+    desc="Running tokenizer on dataset",
+)
+```
+
+Create a training and evaluation [`DataLoader`](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader), and set `pin_memory=True` to speed up data transfer to the GPU during training if your dataset samples are on a CPU.
+
+```py
+train_ds = processed_ds["train"]
```
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:96]: Let's add the imports: 

```
from torch.utils.data import DataLoader
from transformers import default_data_collator
```

File: docs/source/task_guides/ia3.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+<Tip>
+
+Call the [`~PeftModel.print_trainable_parameters`] method to compare the number of trainable parameters of [`PeftModel`] versus the number of parameters in the base model!
+
+</Tip>
+
+Once the configuration is setup, pass it to the [`get_peft_model`] function along with the base model to create a trainable [`PeftModel`].
+
+```py
+from peft import IA3Config
```
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:130]:
```suggestion
from peft import IA3Config, get_peft_model
```

File: docs/source/task_guides/ia3.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+model.print_trainable_parameters()
+"trainable params: 282,624 || all params: 1,229,863,936 || trainable%: 0.022980103060766553"
+```
+
+### Training
+
+Set up an optimizer and learning rate scheduler.
+
+```py
+lr = 8e-3
```
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:143]: let's add the imports:

```
import torch
from transformers import get_linear_schedule_with_warmup
```

File: docs/source/task_guides/ia3.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    optimizer=optimizer,
+    num_warmup_steps=0,
+    num_training_steps=(len(train_dataloader) * num_epochs),
+)
+```
+
+Move the model to the GPU and create a training loop that reports the loss and perplexity for each epoch.
+
+```py
+model = model.to("cuda")
```
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:157]: This block misses `from tqdm import tqdm` and `device = "cuda"`.

File: docs/source/task_guides/ia3.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+```
+
+## Inference
+
+To load the model for inference, use the [`~AutoPeftModelForSeq2SeqLM.from_pretrained`] method. Let's also load a sentence of financial news from the dataset to generate a sentiment for.
+
+```py
+from peft import AutoPeftModelForSeq2SeqLM
+
+model = AutoPeftModelForSeq2SeqLM.from_pretrained("stevhliu/mt0-large-ia3").to("cuda")
```
Comment (BenjaminBossan) [inline docs/source/task_guides/ia3.md:211]: Should "stevhliu" be replaced here by `<your-hf-account-name>` for consistency with the snippet above?

**Review by BenjaminBossan — CHANGES_REQUESTED** [review]
Thanks a lot for adding the IA³ guide. Overall, it looks good. I would suggest to add just a couple of more lines of code to make it so that users can just copy and paste the code and it runs. WDYT?

**Review by BenjaminBossan — APPROVED** [review]
Thanks for making the adjustments quickly. LGTM.
